string_validator.rb

```diff
@@ -0,0 +1,35 @@
+class StringValidator
+
+  PAIRS = {
+    "]" => "[",
+    ")" => "(",
+    "}" => "{"
+  }
+
+  def validate(string)
```

Comment:
Yeah agreed it would be much cleaner. This is pretty messy. I need to spend some time looking at an iterative approach - and probably break it down into a few different methods.